### PR TITLE
feat: Additional heuristics for Java type inference

### PR DESCRIPTION
**Pull request by nmote**
We still don't intend to embed the entire Java standard library, but I've added a few more common methods that will be useful heuristics for type inference. I've also added a few more tests to spot check them.

Test plan: Automated tests

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by nmote** [comment]
Looks like #7942 still hasn't gone out, and the changelog entry for that basically covers what I've done here, too.

File: src/core/Type.ml [hunk 1]
```diff
@@ -157,20 +157,23 @@ let builtin_type_of_string _langTODO str =
...
-  | _, Float -> mkt ?tok "float"
-  | Lang.Java, String -> mkt ?tok "String"
-  | _, String -> mkt ?tok "string"
-  | Lang.Java, Bool -> mkt ?tok "boolean"
-  | _, Bool -> mkt ?tok "bool"
+  | _, Int -> "int"
+  | _, Float -> "float"
+  | Lang.Java, String -> "String"
+  | _, String -> "string"
+  | Lang.Java, Bool -> "boolean"
```
Comment (IagoAbal) [inline src/core/Type.ml:167]: Why `"boolean"` here and not `"bool"` or `"Boolean"` (capitalized)? Why no special case for Java+`Int` ? The special cases probably deserve some comment.

**Review by IagoAbal — APPROVED** [review]
Nice!

**Comment by IagoAbal** [comment]
I'll be merging this since I have to refactor some of this code I think.

File: src/core/Type.ml (hunk 1 above)
Comment (nmote) [reply src/core/Type.ml:167]: The primitive boolean type in Java is called `boolean`. The primitive integer type in Java is called `int`. So that's why.

**Pull request merged**